src/main/java/org/reactome/addlinks/referencecreators/SimpleReferenceCreator.java

```diff
@@ -130,19 +126,25 @@ public void createIdentifiers(long personID, Map<String, T> mapping, List<GKInst
 						}
 					}
 
-					String targetRefDBIdentifier = (String)mapping.get(sourceReferenceIdentifier);
-					this.logger.trace("{} ID: {}; {} ID: {}", this.sourceRefDB, sourceReferenceIdentifier, this.targetRefDB, targetRefDBIdentifier);
-					// Look for cross-references.
-					boolean xrefAlreadyExists = checkXRefExists(sourceReference, targetRefDBIdentifier);
-					if (!xrefAlreadyExists)
-					{
-						this.logger.trace("\tCross-reference {} does not yet exist, need to create a new identifier!", targetRefDBIdentifier);
-						sourceIdentifiersWithNewIdentifier.incrementAndGet();
-						thingsToCreate.add(targetRefDBIdentifier+","+String.valueOf(sourceReference.getDBID())+","+speciesID);
+					// The T value of the mapping can be either a List or a String depending on the FileProcessors invoked.
```

Comment:
You could make both cases use a list of strings with either a single element in the list or multiple so there’s no need to diverge in handling the types differently.

Reply:
I think the possible issue with this is that the method in which this takes place would get modified from the generic T type to a specific type, and since there are a lot of filetypes that feed into this, we might be introducing an untested change for the sake of brevity. I can create an issue to test a refactored version of this code

Reply:
Okay that makes sense.  In this class’ method, though only String or List of String is permitted in the code block line 131.  Maybe the code that uses this class only allows those two types, so I thought if it accepts only List of Strings you could remove the check.

Reply:
Yea its plausible it would be that simple, but I'd feel better addressing this outside of release (and this code needs to be merged for release)

Reply:
Okay - maybe we could open an issue for later?